### PR TITLE
show spinner via goroutine

### LINKED ISSUE #602: Show more activity during deployments
I feel like to someone who doesn't want that much detail, a progress bar would help keep the audience engaged.

### PR DESCRIPTION
**Pull request by nullfunc**
fixes #602 
Spinner currently is tied to main thread timing which may cause sporadic spinner animation (or none at all).

### CODE REVIEW

File: src/pkg/spinner/spinner.go [hunk 1]
```diff
@@ -15,3 +21,21 @@ func (s *Spinner) Next() string {
 	runes := []rune(SpinnerChars)
 	return string([]rune{runes[s.cnt%len(runes)], '\b'})
 }
+
+func (s *Spinner) Start(ctx context.Context) (context.Context, context.CancelFunc) {
```
Comment (lionello) [inline src/pkg/spinner/spinner.go:25]: first return value is never used; remove?

File: src/pkg/spinner/spinner.go [hunk 2]
```diff
@@ -1,5 +1,11 @@
 package spinner
 
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
 var SpinnerChars = `-\|/`
```
Comment (lionello) [inline src/pkg/spinner/spinner.go:9]: `const` or perhaps we can find a sexier spinner and use that? https://stackoverflow.com/questions/2685435/cooler-ascii-spinners

File: src/pkg/cli/tail.go [hunk 3]
```diff
@@ -213,6 +213,10 @@ func tail(ctx context.Context, client client.Client, params TailOptions) error {
 		if doSpinner {
 			term.HideCursor()
 			defer term.ShowCursor()
+
+			var cancelSpinner context.CancelFunc
+			_, cancelSpinner = spin.Start(ctx)
```
Comment (lionello) [inline src/pkg/cli/tail.go:218]:
```suggestion
			cancelSpinner := spin.Start(ctx)
```

File: src/pkg/cli/cert.go [hunk 4]
```diff
@@ -177,11 +167,15 @@ func waitForTLS(ctx context.Context, domain string) error {
 	defer cancel()
 
 	doSpinner := term.StdoutCanColor() && term.IsTerminal()
+	var spin *spinner.Spinner
```
Comment (lionello) [inline src/pkg/cli/cert.go:170]: we can probably remove this

File: src/pkg/cli/cert.go [hunk 5]
```diff
@@ -142,24 +142,14 @@ func generateCert(ctx context.Context, domain string, targets []string, client c
 
 func triggerCertGeneration(ctx context.Context, domain string) error {
 	doSpinner := term.StdoutCanColor() && term.IsTerminal()
+	var spin *spinner.Spinner
```
Comment (lionello) [inline src/pkg/cli/cert.go:145]: we can probably remove this

File: src/pkg/cli/cert.go [hunk 6]
```diff
@@ -142,24 +142,14 @@ func generateCert(ctx context.Context, domain string, targets []string, client c
...
 	if doSpinner {
-		spinCtx, cancel := context.WithCancel(ctx)
+		term.HideCursor()
+		defer term.ShowCursor()
+
+		spin = spinner.New()
+		_, cancel := spin.Start(ctx)
```
Comment (lionello) [inline src/pkg/cli/cert.go:151]: pick one: `cancelSpinner`, `spinnerCancel`, `cancel`.

File: src/pkg/cli/cert.go [hunk 7]
```diff
@@ -211,11 +202,16 @@ func waitForCNAME(ctx context.Context, domain string, targets []string, client c
 	serverSideVerified := false
 	serverVerifyRpcFailure := 0
 	doSpinner := term.StdoutCanColor() && term.IsTerminal()
+	var spin *spinner.Spinner
```
Comment (lionello) [inline src/pkg/cli/cert.go:205]: we can probably remove this

File: src/pkg/spinner/spinner.go (hunk 2 above)
Comment (nullfunc) [reply src/pkg/spinner/spinner.go:9]: I have a stash on my local with this stuff. I'll put up a separate PR with ability to select spinners.

**Pull request merged**